windowsforms/matching-game/net5-windows/vb/MatchingGame/MainForm.vb

```diff
@@ -0,0 +1,91 @@
+﻿Imports MatchingGame.Logic
+Public Class MainForm
+
+    Private _game As Game
+
+    Private Sub StartNewGame()
+
+        _game = Game.Create()
+
+        For i = 0 To tableLayoutPanel1.Controls.Count - 1
+            Dim label = DirectCast(tableLayoutPanel1.Controls(i), Label)
+            Dim column = tableLayoutPanel1.GetColumn(label)
+            Dim row = tableLayoutPanel1.GetRow(label)
+
+            label.Text = _game.GetCard(column, row).ToString()
+            label.ForeColor = label.BackColor
+        Next
+
+        UpdateCards()
+
+    End Sub
+
+    Private Sub UpdateCards()
+        For w = 0 To _game.Width - 1
+            For h = 0 To _game.Height - 1
+                Dim label = DirectCast(tableLayoutPanel1.GetControlFromPosition(w, h), Label)
+                label.ForeColor = IIf(_game.IsOpen(w, h), Color.Black, label.BackColor)
+            Next
+        Next
+    End Sub
+
+    Private Sub label_Click(sender As Object, e As EventArgs) Handles label16.Click, label15.Click, label9.Click, label8.Click, label7.Click, label6.Click, label5.Click, label4.Click, label3.Click, label2.Click, label14.Click, label13.Click, label12.Click, label11.Click, label10.Click, label1.Click
```

Comment:
Re-ordering the labels helps to confirm that no label is missed when reading the code.